connect/api/src/main/java/org/apache/kafka/connect/data/Values.java

```diff
@@ -0,0 +1,1168 @@
+     * @return the representation as an integer, or null if the supplied value was null
+     * @throws DataException if the value could not be converted to an integer
+     */
+    public static Integer convertToInteger(Schema schema, Object value) throws DataException {
+        return (Integer) convertTo(Schema.OPTIONAL_INT32_SCHEMA, schema, value);
+    }
+
+    /**
+     * Convert the specified value to an {@link Type#INT64} long value.
+     * Not supplying a schema may limit the ability to convert to the desired type.
+     *
+     * @param schema the schema for the value; may be null
+     * @param value  the value to be converted; may be null
+     * @return the representation as a long, or null if the supplied value was null
+     * @throws DataException if the value could not be converted to a long
+     */
+    public static Long convertToLong(Schema schema, Object value) throws DataException {
+        return (Long) convertTo(Schema.OPTIONAL_INT64_SCHEMA, schema, value);
+    }
+
+    /**
+     * Convert the specified value to an {@link Type#FLOAT32} float value.
+     * Not supplying a schema may limit the ability to convert to the desired type.
+     *
+     * @param schema the schema for the value; may be null
+     * @param value  the value to be converted; may be null
+     * @return the representation as a float, or null if the supplied value was null
+     * @throws DataException if the value could not be converted to a float
+     */
+    public static Float convertToFloat(Schema schema, Object value) throws DataException {
+        return (Float) convertTo(Schema.OPTIONAL_FLOAT32_SCHEMA, schema, value);
+    }
+
+    /**
+     * Convert the specified value to an {@link Type#FLOAT64} double value.
+     * Not supplying a schema may limit the ability to convert to the desired type.
+     *
+     * @param schema the schema for the value; may be null
+     * @param value  the value to be converted; may be null
+     * @return the representation as a double, or null if the supplied value was null
+     * @throws DataException if the value could not be converted to a double
+     */
+    public static Double convertToDouble(Schema schema, Object value) throws DataException {
+        return (Double) convertTo(Schema.OPTIONAL_FLOAT64_SCHEMA, schema, value);
+    }
+
+    /**
+     * Convert the specified value to an {@link Type#STRING} value.
+     * Not supplying a schema may limit the ability to convert to the desired type.
+     *
+     * @param schema the schema for the value; may be null
+     * @param value  the value to be converted; may be null
+     * @return the representation as a string, or null if the supplied value was null
+     */
+    public static String convertToString(Schema schema, Object value) {
+        return (String) convertTo(Schema.OPTIONAL_STRING_SCHEMA, schema, value);
+    }
+
+    /**
+     * Convert the specified value to an {@link Type#ARRAY} value.
```

Comment:
Updated the doc of this and other methods to hopefully be more clear. Basically, with `convertToList`, `convertToMap`, and `convertToStruct`, the schema is not used, but it is used in many of the numeric conversions when the supplied values are logical values. I kept it in the signature (a) to be consistent with the other forms, and (b) in case we need it for more complex conversions in the future.